src/targets/gpu/include/migraphx/gpu/convert.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_CONVERT_HPP
+#define MIGRAPHX_GUARD_RTGLIB_CONVERT_HPP
+
+#include <migraphx/shape.hpp>
+#include <migraphx/op/convert.hpp>
+#include <migraphx/gpu/oper.hpp>
+#include <migraphx/gpu/device/convert.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+struct context;
+
+struct hip_convert : unary_device<hip_convert, device::convert>
+{
+    op::convert op;
+
+    hip_convert(const op::convert& oper) : op(oper) {}
```

Comment:
This should have a default constructor.

Reply:
This can just be taken by value and then moved.

Reply:
fixed